unittests/compiler/subsortmap.cpp

```diff
@@ -0,0 +1,50 @@
+#include <boost/test/unit_test.hpp>
+#include <kllvm/ast/AST.h>
+
+using namespace kllvm;
+
+BOOST_AUTO_TEST_SUITE(SubsortMapTest)
+
+BOOST_AUTO_TEST_CASE(insert_contains) {
+  SortSet sortSet;
+  SymbolSet symbolSet;
+  auto subsortMap = SubsortMap{};
+
+  auto sort = kore_sort_variable::create("Sort");
+  auto subsort = kore_sort_variable::create("Subsort 1");
+  auto subsort2 = kore_sort_variable::create("Subsort 2");
+
+  subsortMap[sort.get()].insert(subsort.get());
+  subsortMap[sort.get()].insert(subsort2.get());
+
+
+  BOOST_CHECK(subsortMap[sort.get()].contains(subsort.get()));
+  BOOST_CHECK(subsortMap[sort.get()].contains(subsort2.get()));
+  BOOST_CHECK(subsortMap[sort.get()].size() == 2);
+  BOOST_CHECK(subsortMap.size() == 1);
+  BOOST_CHECK(!subsortMap[sort.get()].contains(sort.get()));
+  BOOST_CHECK(!subsortMap[sort.get()].empty());
+}
```

Comment:
I'm not sure what this test is really for; it boils down to checking that `std::unordered_map` works as documented. It's not wrong per se, but I don't think that the test convinces us of anything we didn't already understand.

Reply:
It was meant to test the hashing stuff we suspected to be an error. The idea was to have a very straightforward test indeed. But I can remove it if we decide it doesn't aggregate much as a regression test.

Reply:
The hashing problem we discussed is that if you run:
```
auto s_1 = kore_sort_variable::create("X");
auto s_2 = kore_sort_variable::create("X");
```
then `s_1.get()` and `s_2.get()` are different pointers, so will map to different things.

The solution if this becomes an issue (we don't currently implement this, not sure why) is to have a symbol table mapping sort names to canonical instances.

Reply:
It seems that this is not the case! Please check the latest commit of this PR!